src/VisualStudio/Core/Def/InheritanceMargin/InheritanceMarginTaggerProvider.cs

```diff
@@ -52,6 +54,15 @@ internal sealed class InheritanceMarginTaggerProvider : AsynchronousViewportTagg
 
         protected override TaggerDelay EventChangeDelay => TaggerDelay.OnIdle;
 
+        protected override bool CanCreateTagger(ITextView textView, ITextBuffer buffer)
+        {
+            // Match criterion InheritanceMarginViewMarginProvider uses to determine whether
+            //  the margin is created for the view.
+            var document = textView.TextBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
```

Comment:
Is this only going to be called once for the document on creation? Because right now we're pretty synchronous about ensuring the document is wired up ASAP, but if we make things more async this could break.

Reply:
My understanding is that taggerproviders can/will get called multiple times. If the margin is indeed created, this will definitely get called at that point and succeed.

Reply:
I validated that if I had the CreateTagger return false until the point where the margin was created, and then allowed the CreateTagger to return true, the inheritance margin was still displayed and showed class inheritance information properly.